influxdb3_write/src/cache.rs

```diff
@@ -0,0 +1,132 @@
+use crate::persister::serialize_to_parquet;
+use crate::persister::Error;
+use crate::ParquetFile;
+use bytes::Bytes;
+use datafusion::execution::memory_pool::MemoryPool;
+use datafusion::physical_plan::SendableRecordBatchStream;
+use object_store::memory::InMemory;
+use object_store::path::Path as ObjPath;
+use object_store::ObjectStore;
+use parking_lot::Mutex;
+use std::collections::HashMap;
+use std::sync::Arc;
+
+type MetaData = Mutex<HashMap<String, HashMap<String, HashMap<String, ParquetFile>>>>;
```

Comment:
I'd wrap this in a RWLock instead as queries will hit this a ton, but writes shouldn't be that often.